packages/financial-templates-lib/test/price-feed/CryptoWatchPriceFeed.js

```diff
@@ -184,16 +194,122 @@ describe("CryptoWatchPriceFeed.js", function () {
       0, // Default TWAP length
       60 // Overridden historicalTimestampBuffer of 60 seconds.
     );
-    // Inject data.
-    networker.getJsonReturns = [...validResponses];
+    // Inject data with a missing data point.
+    networker.getJsonReturns = [
+      {
+        result: {
+          60: [
+            [
+              1588376400, // CloseTime
+              1.1, // OpenPrice
+              1, // HighPrice
+              1, // LowPrice
+              1.2, // ClosePrice
+              1, // Volume
+              1, // QuoteVolume
+            ],
+            [1588376460, 1.2, 1, 1, 1.3, 1, 1],
+            [1588376520, 1.3, 1, 1, 1.4, 1, 1],
+            // Missing price periods:
+            // [1588376580, 1.5, 1, 1, 1.6, 1, 1],
+            // [1588376640, 1.6, 1, 1, 1.7, 1, 1],
+            [1588376700, 1.7, 1, 1, 1.8, 1, 1],
+          ],
+        },
+      },
+      { result: { price: 1.5 } },
+    ];
 
     await cryptoWatchPriceFeed.update();
 
-    // Before period 1 should succeed when accounting for historical timestamp buffer.
+    // An input timestamp more than the buffer length before the first period should throw an error.
```

Comment:
thank you for adding these detailed comments. They really helped with understanding how this was wrong and in following your solution.